gui/default/syncthing/device/editDeviceModalView.html

```diff
@@ -10,19 +10,20 @@
         <div id="device-general" class="tab-pane in active">
           <div ng-if="!editingDeviceDefaults()" class="form-group" ng-class="{'has-error': deviceEditor.deviceID.$invalid && deviceEditor.deviceID.$dirty}" ng-init="loadFormIntoScope(deviceEditor)">
             <label translate for="deviceID">Device ID</label>
-            <div ng-if="!editingDeviceExisting()">
-              <div class="input-group">
-                <input name="deviceID" id="deviceID" class="form-control text-monospace" type="text" ng-model="currentDevice.deviceID" required="" valid-deviceid list="discovery-list" aria-required="true" />
-                <div class="input-group-btn">
-                  <button type="button" class="btn btn-default" data-toggle="modal" data-target="#idqr" ng-disabled="!deviceEditor.deviceID.$valid">
-                    <span class="fas fa-qrcode"></span>&nbsp;<span translate>Show QR</span>
-                  </button>
-                </div>
+            <div class="input-group">
+              <input ng-if="currentDevice._editing == 'add'" name="deviceID" id="deviceID" class="form-control text-monospace" type="text" ng-model="currentDevice.deviceID" required="" valid-deviceid list="discovery-list" aria-required="true" />
+              <div ng-if="currentDevice._editing != 'add'" name="deviceID" id="deviceID" class="well well-sm form-control text-monospace" ng-model="currentDevice.deviceID" valid-deviceid style="height: auto;" select-on-click>{{currentDevice.deviceID}}</div>
```

Comment:
Thanks for all the checking. I want to say this clearly though. There is no blind copy & paste here.

These attributes, i.e. `name`, `id`, `ng-model`, and `ng-model` are needed in order for the QR code validation to work. This is how the button is reused in the current iteration of the code. Without the attributes, the ID within the `div` is always "invalid", and as a result, the QR code button is also always greyed out in those instances.

Reply:
Yeah, but if it's read-only anyway, there is no point in checking validity.  The user has no control over it, and we won't ever get an invalid device ID from a pending or existing device.

Reply:
That's true, but is there a better way that would get rid of the validation and still be able to use a single QR code button for everything?

I'd really like not to have duplicated code related to the button, especially since I'm also working on the other PR that adds a "copy to clipboard" and "send by email" buttons. I think the code would be way easier to maintain if only a single instance of the buttons existed (rather than having one with validation, and another one without it for every button).

Reply:
As mentioned below, you need to adjust the `ng-disabled` attribute.

    ng-disabled="currentDevice._editing == 'add' && !deviceEditor.deviceID.$valid"

Or using the suggested wrapper function:

    ng-disabled="editingDeviceAdditional() && !deviceEditor.deviceID.$valid"

Reply:
This makes sense. Thank you very much 🙂. I'll try to implement all the fixes and then come back, probably sometime in the next few days.